src/pykeen/nn/functional.py

```diff
@@ -1294,3 +1295,36 @@ def cp_interaction(
         r,
         t,
     )
+
+
+def triple_re_interaction(
+    # head
+    h: torch.FloatTensor,
+    # relation
+    r_head: torch.FloatTensor,
+    r_mid: torch.FloatTensor,
+    r_tail: torch.FloatTensor,
+    # tail
+    t: torch.FloatTensor,
+    # version 2: relation factor offset
+    u: Optional[float] = None,
+    # extension: negative (power) norm
+    p: int = 2,
+    power_norm: bool = False,
+) -> torch.FloatTensor:
```

Comment:
please add documentation, including a beautiful math descirpton in latex

Reply:
7f130f0